src/gc/gc_manager.cpp

```diff
@@ -23,39 +23,40 @@ namespace peloton {
 namespace gc {
 
 // Check a tuple and reclaim all varlen field
-void GCManager::CheckAndReclaimVarlenColumns(storage::TileGroup *tg, oid_t tuple_id) {
-    oid_t tile_count = tg->tile_count;
-    oid_t tile_col_count;
-    type::TypeId type_id;
-    char *tuple_location;
-    char *field_location;
-    char *varlen_ptr;
-
-      for (oid_t tile_itr = 0; tile_itr < tile_count; tile_itr++) {
-        const catalog::Schema &schema = tg->tile_schemas[tile_itr];
-
-        tile_col_count = schema.GetColumnCount();
-
-        storage::Tile *tile = tg->GetTile(tile_itr);
-        PELOTON_ASSERT(tile);
-        for (oid_t tile_col_itr = 0; tile_col_itr < tile_col_count; ++tile_col_itr) {
-            type_id = schema.GetType(tile_col_itr);
-
-              if ((type_id != type::TypeId::VARCHAR && type_id != type::TypeId::VARBINARY)
-                             || (schema.IsInlined(tile_col_itr) == true)) {
-                // Not of varlen type, or is inlined, skip
-                  continue;
-              }
-            // Get the raw varlen pointer
-              tuple_location = tile->GetTupleLocation(tuple_id);
-            field_location = tuple_location + schema.GetOffset(tile_col_itr);
-            varlen_ptr = type::Value::GetDataFromStorage(type_id, field_location);
-            // Call the corresponding varlen pool free
-              if (varlen_ptr != nullptr) {
-                tile->pool->Free(varlen_ptr);
-              }
-          }
+void GCManager::CheckAndReclaimVarlenColumns(storage::TileGroup *tile_group,
+                                             oid_t tuple_id) {
+  oid_t tile_count = tile_group->tile_count_;
+  oid_t tile_col_count;
```

Comment:
tile_count and tile_col_count should be uint32_t not oid_t.
While use of oid_t will compile and run correctly, it is confusing and incorrect style.